identity-service/src/notifications/index.js

```diff
@@ -440,6 +444,39 @@ class NotificationProcessor {
   }
 }
 
+/**
+ * Filters out notifications whose initiators are deemed abusive.
+ * @param {Object[]} notifications
+ * @returns {Promise<Object[]>} notifications - after having filtered out notifications from bad initiators
+ */
+async function filterOutAbusiveUsers (notifications) {
+  const notificationsWithAbuse = await Promise.all(
+    notifications.map(async notification => ({
```

Comment:
this is bad. we should pull all the ids and do one query

Reply:
yup!